### PR TITLE
Improve UI customizability

### PR DESCRIPTION
**Pull request by mauriziopinotti**
Trying to reach a good trade-off between library consistency and user customizability. Sticky area and week/day headers now have basic customizations (colors and widget builders) and more can be added in the future.

Flutter Demo

Example:

```dart
              weekHeaderBuilder: (context, weekDate) => GanttChartDefaultWeekHeader(
                      weekDate: weekDate,
                      color: Colors.black,
                      backgroundColor: Colors.yellow,
                      border: const BorderDirectional(
                        end: BorderSide(color: Colors.green),
                      )),
```

Same can be done for dayHeader and StickyArea.

### CODE REVIEW

File: lib/src/gantt_default_week_header.dart [hunk 1]
```diff
@@ -1,41 +1,56 @@
...
   }
+
+  Color get defaultColor => Colors.white;
+
+  Color get defaultBackgroundColor => Colors.blue.shade900;
+
+  BoxBorder get defaultBorder => const BorderDirectional(
+        start: BorderSide(),
+        bottom: BorderSide(),
+      );
```
Comment (ahmednfwela) [inline lib/src/gantt_default_week_header.dart:38]: can these be defined as static consts ?

File: lib/src/gantt_default_day_header.dart [hunk 2]
```diff
@@ -3,36 +3,46 @@ import 'package:flutter/material.dart';
...
       ),
     );
   }
+
+  Color defaultColor(bool isHoliday) => isHoliday ? Colors.white : Colors.black;
+
+  Color defaultBackgroundColor(bool isHoliday) =>
+      isHoliday ? Colors.grey.shade800 : Colors.white;
+
+  BoxBorder get defaultBorder => const BorderDirectional(
```
Comment (ahmednfwela) [inline lib/src/gantt_default_day_header.dart:40]: also please define this as static const

**Review by ahmednfwela — COMMENTED** [review]
Thanks for this great PR, indeed this should be a good balance between defaults and customizability, would you please update the changelog to include breaking changes, and update pubspec with new major version

File: lib/src/gantt_default_week_header.dart (hunk 1 above)
Comment (ahmednfwela) [reply lib/src/gantt_default_week_header.dart:38]: or they can be defined as `static Color defaultColor  = Colors.white;`
which will allow developers to do 
`GanttChartDefaultWeekHeader.defaultColor = SomeColor;`

**Comment by mauriziopinotti** [comment]
Sure, I'll cleanup the code as you requested in the previous comments then update the changelog. BTW, it should be 100% backward compatible :)

**Comment by ahmednfwela** [comment]
> Sure, I'll cleanup the code as you requested in the previous comments then update the changelog. BTW, it should be 100% backward compatible :)

the only breaking change i can see is the `isHoliday` parameter changing from a Function to a bool

File: lib/src/gantt_default_day_header.dart [hunk 3]
```diff
@@ -3,36 +3,49 @@ import 'package:flutter/material.dart';
 import 'week_day.dart';
 
 class GanttChartDefaultDayHeader extends StatelessWidget {
+  final Color? color;
+  final Color? backgroundColor;
```
Comment (ahmednfwela) [inline lib/src/gantt_default_day_header.dart:7]: I think that these should be defined as functions that take `isHoliday` as parameter

Comment (ahmednfwela) [reply lib/src/gantt_default_day_header.dart:7]: or not , since it's passed as a `bool` now, sorry

**Comment by mauriziopinotti** [comment]
You're right, I thought `GanttChartDefaultDayHeader` was for internal use only before this PR but actually it was already exposed to the consumer.

I've updated the patch with your suggestions. Should the new version be 1.0.0 or 0.2.0? I didn't want to steal the honor of bumping to 1.0.0 so I opted for 0.2.0.

File: lib/src/gantt_default_day_header.dart (hunk 3 above)
Comment (mauriziopinotti) [reply lib/src/gantt_default_day_header.dart:12]: > I think that these should be defined as functions that take `isHoliday` as parameter

Yeah, I moved the isHoliday logic to the consumer but it makes sense both ways to me...

File: lib/src/gantt_default_day_header.dart [hunk 4]
```diff
@@ -3,36 +3,55 @@ import 'package:flutter/material.dart';
 import 'week_day.dart';
 
 class GanttChartDefaultDayHeader extends StatelessWidget {
+  static const defaultColor = Colors.black;
+  static const defaultBackgroundColor = Colors.white;
+  static const defaultHolidayColor = Colors.white;
+  static final defaultHolidayBackgroundColor = Colors.grey.shade800;
```
Comment (ahmednfwela) [inline lib/src/gantt_default_day_header.dart:9]: what do you think about removing `const/final` so they can be changed later by the developers, e.g.:
`GanttChartDefaultDayHeader.defaultColor = SomeColor;`
might be useful when dealing with multiple gantt charts on the same app

**Comment by ahmednfwela** [comment]
> Should the new version be 1.0.0 or 0.2.0?

I believe we should stay away from 1.0.0 until the library is mature enough

File: lib/src/gantt_default_day_header.dart (hunk 4 above)
Comment (mauriziopinotti) [reply lib/src/gantt_default_day_header.dart:9]: > what do you think about removing `const/final` so they can be changed later by the developers, e.g.: `GanttChartDefaultDayHeader.defaultColor = SomeColor;` might be useful when dealing with multiple gantt charts on the same app

Uhm, IMHO those are like "platform defaults" which should not be changed, they represent "standard" colors. Developers can still pass colors as arguments, so it should be flexible enough.

Comment (ahmednfwela) [reply lib/src/gantt_default_day_header.dart:9]: ok then, I think PR should be fine now

Comment (ahmednfwela) [reply lib/src/gantt_default_day_header.dart:9]: or do you have any additional changes in mind ?

Comment (mauriziopinotti) [reply lib/src/gantt_default_day_header.dart:9]: I think I'm done (for the time being) :smile:

**Comment by ahmednfwela** [comment]
a new PR (#8) and version was published on pub.dev, thanks for your contribution

**Pull request merged**